packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -356,11 +356,10 @@ class _TextLayout {
     // Luckily they have the same bidi embedding level as the paragraph as per
     // https://unicode.org/reports/tr9/#L1, so we can anchor the caret to the
     // last logical trailing space.
-    final bool hasTrailingSpaces = switch (rawString.codeUnitAt(rawString.length - 1)) {
-      0x9 ||        // horizontal tab
-      0x3000 ||     // ideographic space
-      0x20 => true, // space
-      _ => false,
+    final String lastCharactor = rawString[rawString.length - 1];
+    final bool hasTrailingSpaces = switch (lastCharactor.codeUnitAt(0)) {
```

Comment:
There's little point in using a switch expression now I think? This could just be an `||`.